File: fix_slashes.py
import os
import re

directory = r'c:\Users\ALlahabi\Desktop\cmder\Mojaz\src\frontend\src\services'
# Pattern to match: axios.get('/path' or axios.get(`/path`
# Also handles apiClient.post etc.
pattern = re.compile(r"(\.(get|post|patch|put|delete)\s*\(\s*)(['\"`])/([^/][^'\"`]*?)(['\"`])")

for filename in os.listdir(directory):
    if filename.endswith('.ts'):
        filepath = os.path.join(directory, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        new_content = pattern.sub(r'\1\3\4\5', content)
        
        if new_content != content:
            print(f"Updating {filename}")
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
        else:
            print(f"Skipping {filename}")

File: fix_slashes.js
const fs = require('fs');
const path = require('path');

const directory = 'c:/Users/ALlahabi/Desktop/cmder/Mojaz/src/frontend/src/services';
// Added support for TS generics like .post<ApiResponse<T>>
const pattern = /(\.(get|post|patch|put|delete)(?:<[\s\S]*?>)?\s*\(\s*)(['"`])\/([^\/][^'"`]*?)(['"`])/g;

fs.readdirSync(directory).forEach(file => {
  if (file.endsWith('.ts')) {
    const filePath = path.join(directory, file);
    const content = fs.readFileSync(filePath, 'utf8');
    const newContent = content.replace(pattern, '$1$3$4$5');
    
    if (newContent !== content) {
      console.log(`Updating ${file}`);
      fs.writeFileSync(filePath, newContent, 'utf8');
    } else {
      console.log(`Skipping ${file}`);
    }
  }
});
